lib/adrian/directory_queue.rb

```diff
@@ -36,8 +37,13 @@ def pop_item
 
     def push_item(value)
       item = wrap_item(value)
-      item.move(available_path)
-      item.touch
+      if @touch_first
+        item.touch
+        item.move(available_path)
+      else
+        item.move(available_path)
+        item.touch
+      end
       self
```

Comment:
Is there a reason why we don't `rescue` `Errno::ENOENT` as we do in `DirectoryQueue#reserve`?

Reply:
If not, I'd like to propose changing the following methods to the code below:

```ruby
def pop_item
  while item = items.shift
    return item if reserve(item, reserved_path)
  end
end

def push_item(value)
  item = wrap_item(value)
  reserve(item, available_path)

 self
end

def reserve(item, path)
  if @touch_first
    item.touch
    item.move(path)
  else
    item.move(path)
    item.touch
  end
  
  true
rescue Errno::ENOENT => e
  false
 end
```

Reply:
@rodrigoalvesvieira I think when we call `reserve` there is a potential for the item to have already been reserved, so the rescue prevents us from raising an error trying to pop an item off the queue. That scenario is less likely when we call `push`, since we are moving a file to a directory whose existence is ensured by `mkdir_p`, but there are good arguments to be made about keeping things dry.